### DISCUSSION TITLE
Unable to Download Agents from On-Prem Server

### DISCUSSION THREAD

**Discussion by modem-19** [amidaware/tacticalrmm#1916] [General]
I just noticed this today with my testing, that when I tried to download the RMM/mesh agent from my on prem server, I get the message:  "Something went wrong.  Check debug error log for exact error message.  400:  Bad Request"  See the attached photo.

My test RMM setup is on a Hyper-V Ubuntu 22.04 setup.  I have not made any OS upgrades or changes as advised by the documentation.  

Anyone have any idea what might be happening?

![rmm-error](https://github.com/user-attachments/assets/3222a4e9-7a90-4a32-8260-617b255d3a6a)

**Comment by silversword411** [comment]
Any 3rd party av/security slicing connections on download?

**Reply by modem-19** [reply]
It doesn't make it as far as the download.  As soon as I click on Agents > Install Agent menu, fill out the options I want, and click on "Generate and download EXE", I immediately get that message.

The other options as the Powershell and Manual download work to download files, but the agents aren't talking back to the server and never get added on the server.

I suspect something here on my testing sandbox server, but I'm not sure where the debug errors are located at to start investigating.

**Comment by modem-19** [comment]
Problem is fixed.  It wasn't RMM, but rather and OS issue and DNS all of a sudden resolving to a non DNS server.

**Reply by N3XPW** [reply]
i got the same error msg on the exact same setup as you (hyper-v and ubuntu 22.04). In my Browser-Console is the following error msg:
{"detail":"Authentication credentials were not provided."}.

Did you have the same?
and/or what exactly did you changed that it worked now?
